### PR TITLE
Release 2.3

### PR DESCRIPTION
**Pull request by WackerO**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/qbic-pipelines/rnadeseq/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: assets/RNAseq_report.Rmd [hunk 1]
```diff
@@ -875,57 +925,145 @@ The read sequencing quality was analyzed using `FASTQC`. More detailed informati
 
 The read mapping and gene assignment statistics, together with other quality control parameters are summarized here.
 
-The table below, extracted from the `MultiQC` report, shows a summary of the bioinformatics analysis quality control.
-Note: Duplicate intercept (%) is the percentage of duplicate reads for the intercept of a linear model of duplicate reads vs read counts. High numbers of duplicates at low read counts can indicate low library complexity with technical duplication."
+The table below, extracted from the `MultiQC` report, shows a summary of the bioinformatics analysis quality control.",
+ifelse(mqc_version == 'old_mqc',
+"Note: Duplicate intercept (%) is the percentage of duplicate reads for the intercept of a linear model of duplicate reads vs read counts. High numbers of duplicates at low read counts can indicate low library complexity with technical duplication.",
+"")
```
Comment (tillenglert) [inline assets/RNAseq_report.Rmd:931]: Is that empty string there for a reason?

File: conf/test_custom_gmt.config [hunk 2]
```diff
@@ -28,6 +28,8 @@ params {
     software_versions = 'https://raw.githubusercontent.com/qbic-pipelines/rnadeseq/dev/testdata/software_versions.csv'
     multiqc = 'https://raw.githubusercontent.com/qbic-pipelines/rnadeseq/dev/testdata/MultiQC.zip'
     run_pathway_analysis = true
+    datasources = 'KEGG,REAC'
     genome = 'GRCm38'
     custom_gmt = 'https://raw.githubusercontent.com/qbic-pipelines/rnadeseq/dev/testdata/gprofiler_full_mmusculus.ENSG.gmt'
+    datasources = 'KEGG,REAC'
```
Comment (tillenglert) [inline conf/test_custom_gmt.config:34]: This row seems to be duplicated?

File: docs/usage.md [hunk 3]
```diff
@@ -89,7 +89,7 @@ The `qbic-pipelines/rnadeseq` pipeline relies on the output from the `nf-core/rn
 The typical command for running the pipeline is as follows:
 
 ```bash
-nextflow run qbic-pipelines/rnadeseq -r 2.0.1 -profile docker \
+nextflow run qbic-pipelines/rnadeseq -r 2.2 -profile docker \
```
Comment (tillenglert) [inline docs/usage.md:92]: shouldn't that be 2.3 then? same below

**Review by tillenglert — APPROVED** [review]
LGTM (within my current knowledge of rnadeseq and R)! Still I had some minor comments 👍

File: assets/RNAseq_report.Rmd (hunk 1 above)
Comment (WackerO) [reply assets/RNAseq_report.Rmd:931]: It's there to make the ifelse work 😅 Not sure if there is a more efficient way of constructing these pasted strings

File: conf/test_custom_gmt.config (hunk 2 above)
Comment (WackerO) [reply conf/test_custom_gmt.config:34]: Oop, you're right!

File: docs/usage.md (hunk 3 above)
Comment (WackerO) [reply docs/usage.md:92]: Ah yes of course, I'll update that

**Review by d4straub — APPROVED** [review]
Looks good to me.
Unfortunately 2 tests are failing.
Two tests (EditroConfig & Prettier) arent required any more and I adapted the settings accordingly.

File: docs/usage.md [hunk 4]
```diff
@@ -157,7 +157,7 @@ curl https://codeload.github.com/qbic-pipelines/rnadeseq/tar.gz/master | tar -xz
 Afterwards, you should be able to also run test_star_rsem and test_star_salmon from qbic-pipelines/rnadeseq without manually cloning, e.g.:
 
 ```bash
-nextflow run qbic-pipelines/rnadeseq -r 2.0.1 -profile docker,test_star_salmon
+nextflow run qbic-pipelines/rnadeseq -r 2.3 -profile docker,test_star_salmon
```
[the remark this answers is not in the document]
Comment (WackerO) [reply docs/usage.md:160]: No, it has the default `results` defined in the `nextflow.config` (that's been like that forever; is that bad?)

[the remark this answers is not in the document]
Comment (d4straub) [reply docs/usage.md:160]: No thats fine, was just wondering.

File: assets/RNAseq_report.Rmd [hunk 5]
```diff
@@ -864,6 +869,51 @@ for (i in resultsNames(cds)[-1]) {
...
+            "Sample",
+            "raw_total_sequences",
+            "reads_mapped_percent",
+            "reads_duplicated_percent"
+        )
+    if (all(columns %in% colnames(mqc_stats))) {
+        mqc_version <- 'new_mqc'
+    } else {
+        stop("Could not find a suitable multiqc table; please provide a correct multiqc.zip file or omit the parameter --multiqc altogether.")
+    }
```
[the remark this answers is not in the document]
Comment (WackerO) [reply assets/RNAseq_report.Rmd:912]: I know, this is indeed not a nice code section but unfortunately I have to do this check twice, once inside a nested ifelse and once in a separate ifelse...

**Comment by WackerO** [comment]
As https://nfcore.slack.com/archives/CE6SDBX2A/p1663077291333929 and https://nfcore.slack.com/archives/C02AL61NXPH/p1651131006534289 describe the same issue with the failing branch protection, and I could not find any solution there (and my attempts to fix it also failed), I'm going to ignore this error and merge.
Maybe it will resolve itself in the future but in this case, it is not a problem IMO.

**Pull request merged**